### PR TITLE
Add force_ascii=False when generating JSONL

### LINKED ISSUE #2046: Non-ASCII characters in testsets and report JSON files
### Issue Type

Bug

### Source

source

### Giskard Library Version

2.15.1

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.12

### Installed python packages

_No response_

### Current Behaviour?

```shell
Right now, the testsets are generated with ASCII characters in the texts. This is not optimal with content in Swedish, for instance.
```

### Standalone code OR list down the steps to reproduce the issue

```shell
Generate a testset in Swedish and/or a report.
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PierreMesure**
## Description

Fixes #2046

## Related Issue

#2046

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [X] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [X] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [X] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [X] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.
- [ ] I've updated the `pdm.lock` running `pdm update-lock` (only applicable when `pyproject.toml` has been
  modified)

### CODE REVIEW

**Review by henchaves — APPROVED** [review]

**Comment by henchaves** [comment]
@PierreMesure that's perfect! Thanks a lot for your contribution.

File: giskard/rag/report.py [hunk 1]
```diff
@@ -156,18 +156,18 @@ def save(self, folder_path: str):
 
         report_details = {"recommendation": self._recommendation}
         with open(path / "report_details.json", "w", encoding="utf-8") as f:
-            json.dump(report_details, f)
+            json.dump(report_details, f, ensure_ascii=False)
 
-        self._knowledge_base._knowledge_base_df.to_json(path / "knowledge_base.jsonl", orient="records", lines=True)
+        self._knowledge_base._knowledge_base_df.to_json(path / "knowledge_base.jsonl", orient="records", lines=True, force_ascii=False)
```
Comment (henchaves) [inline giskard/rag/report.py:161]: making pre-commit test to pass
```suggestion
        self._knowledge_base._knowledge_base_df.to_json(
            path / "knowledge_base.jsonl", orient="records", lines=True, force_ascii=False
        )
```

File: giskard/rag/report.py [hunk 2]
```diff
@@ -156,18 +156,18 @@ def save(self, folder_path: str):
...
         with open(path / "knowledge_base_meta.json", "w", encoding="utf-8") as f:
-            json.dump(self._knowledge_base.get_savable_data(), f)
+            json.dump(self._knowledge_base.get_savable_data(), f, ensure_ascii=False)
 
         with open(path / "agent_answer.json", "w", encoding="utf-8") as f:
-            json.dump([{"message": output.message, "documents": output.documents} for output in self._model_outputs], f)
+            json.dump([{"message": output.message, "documents": output.documents} for output in self._model_outputs], f, ensure_ascii=False)
```
Comment (henchaves) [inline giskard/rag/report.py:166]: making pre-commit test to pass
```suggestion
            json.dump(
                [{"message": output.message, "documents": output.documents} for output in self._model_outputs], 
                f, 
                ensure_ascii=False
            )
```

**Comment by PierreMesure** [comment]
You're welcome, thanks for your work on Giskard!

**Pull request merged**